File: ruby/cookbook/recipe_command.rb
#!/usr/bin/env ruby

module Recipe

    require './cookbook/util/logging'

    class << self

        @@logger = LoggerFactory.getLogger File.basename __FILE__

        def title

            @@logger.debug '# Command'

        end

        def cook

            command = 'ls'

            # コマンド実行

            IO.popen(command) do |pipe|
                # コマンドのpid取得
                pid = pipe.pid
                @@logger.debug "#{command}[##{pid}] started"
                # コマンド実行結果（標準出力）を取得
                pipe.each do |line|
                    @@logger.debug line.chomp
                end
                 @@logger.debug "#{command}[##{pid}] ended"
            end

        end

    end

end

if __FILE__ == $0

    Recipe.title
    Recipe.cook

end

File: python/cookbook/recipe_command.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-

import sys
from subprocess import Popen, PIPE
from cookbook.util.logging import LoggerFactory

logger = LoggerFactory.getLogger(__name__)

def title():

    logger.debug('# Command')

def cook():

    commands = ['ls']

    # コマンド実行

    for command in commands:

        proc = Popen(command.split(' '), stdout=PIPE)
        # コマンドのpid取得
        pid = proc.pid
        logger.debug("%s[#%s] started" % (command, pid))
        # コマンド実行結果（標準出力）を取得
        pipe = proc.stdout
        for line in pipe.readlines():
            logger.debug(line.rstrip().decode(sys.stdin.encoding))
        pipe.close()
        logger.debug("%s[#%s] ended" % (command, pid))

if __name__ == '__main__':

    title()
    cook()
